src/Examples/UnitModels/cstr_testing.ipynb

```diff
@@ -0,0 +1,1123 @@
+    "    - Adding additional constraints \n",
+    "\n",
+    "\n",
+    "## Problem Statement\n",
+    "\n",
+    "This example is adapted from Fogler, H.S., Elements of Chemical Reaction Engineering 5th ed., 2016, Prentice Hall,  p. 157-160.\n",
+    "\n",
+    "Ethylene glycol (EG) is a high-demand chemical, with billions of pounds produced every year for applications such as vehicle anti-freeze. EG may be readily obtained from the hydrolysis of ethylene oxide in the presence of a catalytic intermediate. In this example, an aqueous solution of ethylene oxide hydrolizes after mixing with an aqueous solution of sulfuric acid catalyst:\n",
+    "\n",
+    "**C<sub>2</sub>H<sub>4</sub>O</sub> + H<sub>2</sub>O</sub> + H<sub>2</sub>S</sub>O<sub>4</sub> → C<sub>2</sub>H<sub>6</sub>O<sub>2</sub> + H<sub>2</sub>S</sub>O<sub>4</sub>**\n",
+    "\n",
+    "This reaction often occurs by two mechanisms, as the catalyst may bind to either reactant before the final hydrolysis step; we will simplify the reaction to a single step for this example.\n",
+    "\n",
+    "The flowsheet that we will be using for this module is shown below with the stream conditions. We will be processing ethylene oxide and catalyst solutions of fixed concentrations to produce 200 MM lb/year of EG. As shown in the flowsheet, the process consists of a mixer M101 for the two inlet streams, a heater H101 to preheat the feed to the reaction temperature, and a CSTR unit R101 with zero overall duty (external cooling required for the exothermic reaction). We will assume ideal solutions and thermodynamics for this flowsheet, as well as well-mixed liquid behavior (no vapor phase) in the reactor. The properties required for this module are available in the same directory:\n",
+    "\n",
+    "- egprod_ideal.py\n",
+    "- egprod_reaction.py\n",
+    "\n",
+    "The state variables chosen for the property package are **molar flows of each component by phase in each stream, temperature of each stream and pressure of each stream**. The components considered are: **ethylene oxide, water, sulfuric acid and ethylene glycol** and the process occurs in liquid phase only. Therefore, every stream has 4 flow variables, 1 temperature and 1 pressure variable. \n",
+    "\n",
+    "![](egprod_flowsheet.png)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Importing required pyomo and idaes components\n",
+    "\n",
+    "\n",
+    "To construct a flowsheet, we will need several components from the pyomo and idaes package. Let us first import the following components from Pyomo:\n",
+    "- Constraint (to write constraints)\n",
+    "- Var (to declare variables)\n",
+    "- ConcreteModel (to create the concrete model object)\n",
+    "- Expression (to evaluate values as a function of variables defined in the model)\n",
+    "- Objective (to define an objective function for optimization)\n",
+    "- SolverFactory (to solve the problem)\n",
```

Comment:
Is this needed? I notice that you import `get_solver` later on, and we tend to use that as the default approach in most of our code.